### PR TITLE
Add margin, padding and alignment to tailed layout transformation

### PR DESCRIPTION
**Pull request by WojciechBarczynski**

### CODE REVIEW

File: compositor_render/src/transformations/builtin/params/tiled_layout.rs [hunk 1]
```diff
@@ -73,69 +75,98 @@ impl TiledLayoutParams {
...
-        for row in 0..(rows_cols.rows - 1) {
-            for col in 0..rows_cols.cols {
-                let top_left_corner = (
-                    x_padding + col as f32 * tile_size.width as f32,
-                    y_padding + row as f32 * tile_size.height as f32,
-                );
+        let mut top = additional_top_padding
+            + justified_padding_y
+            + tailed_layout_spec.padding as f32
+            + tailed_layout_spec.margin as f32;
```
Comment (WojciechBarczynski) [inline compositor_render/src/transformations/builtin/params/tiled_layout.rs:102]: This and `left` don't have to be `mut`, but I found modifying it more readable than calculating it in every iteration

File: compositor_render/src/transformations/builtin.rs [hunk 2]
```diff
@@ -39,10 +39,9 @@ impl Builtin {
                 background_color_rgba,
                 ..
             } => Some(rgba_to_wgpu_color(background_color_rgba)),
-            BuiltinSpec::TiledLayout {
-                background_color_rgba,
-                ..
-            } => Some(rgba_to_wgpu_color(background_color_rgba)),
+            BuiltinSpec::TiledLayout(tailed_layout_spec) => Some(rgba_to_wgpu_color(
+                &tailed_layout_spec.background_color_rgba,
+            )),
```
Comment (wkozyra95) [inline compositor_render/src/transformations/builtin.rs:44]:
```suggestion
            BuiltinSpec::TiledLayout(spec) => Some(rgba_to_wgpu_color(
                &spec.background_color_rgba,
            )),
```

File: compositor_render/src/transformations/builtin.rs [hunk 3]
```diff
@@ -60,20 +59,20 @@ impl Builtin {
...
         }
 
-        match self.spec {
-            BuiltinSpec::TransformToResolution { resolution, .. } => resolution,
-            BuiltinSpec::FixedPositionLayout { resolution, .. } => resolution,
-            BuiltinSpec::TiledLayout { resolution, .. } => resolution,
+        match &self.spec {
+            BuiltinSpec::TransformToResolution { resolution, .. } => *resolution,
+            BuiltinSpec::FixedPositionLayout { resolution, .. } => *resolution,
+            BuiltinSpec::TiledLayout(tailed_layout_spec) => tailed_layout_spec.resolution,
```
Comment (wkozyra95) [inline compositor_render/src/transformations/builtin.rs:65]:
```suggestion
            BuiltinSpec::TiledLayout(spec) => spec.resolution,
```

File: compositor_render/src/transformations/builtin.rs [hunk 4]
```diff
@@ -60,20 +59,20 @@ impl Builtin {
...
 
     pub fn resolution_from_spec(&self) -> Option<Resolution> {
-        match self.spec {
-            BuiltinSpec::TransformToResolution { resolution, .. } => Some(resolution),
-            BuiltinSpec::FixedPositionLayout { resolution, .. } => Some(resolution),
-            BuiltinSpec::TiledLayout { resolution, .. } => Some(resolution),
+        match &self.spec {
+            BuiltinSpec::TransformToResolution { resolution, .. } => Some(*resolution),
+            BuiltinSpec::FixedPositionLayout { resolution, .. } => Some(*resolution),
+            BuiltinSpec::TiledLayout(tailed_layout_spec) => Some(tailed_layout_spec.resolution),
```
Comment (wkozyra95) [inline compositor_render/src/transformations/builtin.rs:75]:
```suggestion
            BuiltinSpec::TiledLayout(spec) => Some(spec.resolution),
```

File: compositor_render/src/transformations/builtin/params.rs [hunk 5]
```diff
@@ -50,15 +50,9 @@ impl BuiltinParams {
...
-                resolution: output_resolution,
-                tile_aspect_ratio,
-                ..
-            } => BuiltinParams::TiledLayout(TiledLayoutParams::new(
-                input_resolutions,
-                *tile_aspect_ratio,
-                *output_resolution,
-            )),
+            BuiltinSpec::TiledLayout(tailed_layout_spec) => BuiltinParams::TiledLayout(
+                TiledLayoutParams::new(input_resolutions, tailed_layout_spec),
```
Comment (wkozyra95) [inline compositor_render/src/transformations/builtin/params.rs:54]:
```suggestion
            BuiltinSpec::TiledLayout(spec) => BuiltinParams::TiledLayout(
                TiledLayoutParams::new(input_resolutions, spec),
```

**Review by wkozyra95 — APPROVED** [review]
LGTM (didn't check matho

**Pull request merged**